src/components/ListUniqueValues.vue

```diff
@@ -0,0 +1,222 @@
+<template>
+  <div class="list-unique-values">
+    <div class="list-unique-values__title">Filter values</div>
+    <input class="list-unique-values__search-box" placeholder="Search 50 items" v-model="search" />
+    <div class="list-unique-values__select-clear-all">
+      <span class="list-unique-values__select-all" @click="selectAll">Select all</span> &nbsp;
+      <span class="list-unique-values__clear-all" @click="clearAll">Clear all</span>
+    </div>
+    <div
+      class="list-unique-values__checkbox-container"
+      v-for="option in searchedOptions"
+      :key="option.value"
+    >
+      <CheckboxWidget
+        :label="`${option.value} (${option.count})`"
+        :value="isChecked(option)"
+        @input="toggleCheck(option)"
+      />
+    </div>
+    <div class="list-unique-values__load-all-values" v-if="loaded">
+      <div>List maybe incomplete</div>
+      <div @click="loadAllValues" class="list-unique-values__load-all-values-button">
+        Load all values
+      </div>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+import _difference from 'lodash/difference';
+import _union from 'lodash/union';
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+import { FilterConditionInclusion } from '@/lib/steps.ts';
+
+// This type will be imported from @/lib/dataset/helpers.ts
+type ColumnValueStat = {
+  value: any;
+  count: number;
+};
+
+import CheckboxWidget from './stepforms/widgets/Checkbox.vue';
+@Component({
+  name: 'list-unique-values',
+  components: { CheckboxWidget },
+})
+export default class ListUniqueValues extends Vue {
+  @Prop({
+    type: Array,
+    required: true,
+  })
+  options!: ColumnValueStat[];
+
+  @Prop({
+    type: Object,
+    required: true,
+  })
+  filter!: FilterConditionInclusion;
+
+  @Prop({
+    type: Boolean,
+    required: true,
+  })
+  loaded!: boolean;
+
+  search = '';
+
+  isChecked(option: ColumnValueStat): boolean {
+    if (this.filter.operator == 'in') {
+      return this.filter.value.includes(option.value);
+    } else {
+      return !this.filter.value.includes(option.value);
+    }
+  }
+
+  selectAll() {
+    if (this.search === '') {
+      this.$emit('input', {
+        column: this.filter.column,
+        operator: 'nin',
+        value: [],
+      } as FilterConditionInclusion);
+    } else {
+      if (this.filter.operator === 'in') {
+        this.$emit('input', {
+          column: this.filter.column,
+          operator: 'in',
+          value: _union(
+            this.filter.value,
+            this.searchedOptions.map(option => option.value),
+          ),
+        } as FilterConditionInclusion);
+      } else {
+        this.$emit('input', {
+          column: this.filter.column,
+          operator: 'nin',
+          value: _difference(
+            this.filter.value,
+            this.searchedOptions.map(option => option.value),
+          ),
+        } as FilterConditionInclusion);
+      }
```

Comment:
I think if would be clearer to declare a variable of type FilterConditionInclusion and emit it afterwards:

```suggestion
   const newFilter = {
        column: this.filter.column,
        operator: 'nin',
        value: [],
      };
    if (this.search !== '') {
      // Only select all _searched_ values
      if (this.filter.operator === 'in') {
        newFilter.operator = 'in';
        newFilter.value = _union(
            this.filter.value,
            this.searchedOptions.map(option => option.value),
          );
      } else {
        newFilter.value = _difference(
            this.filter.value,
            this.searchedOptions.map(option => option.value),
          )
       }
       this.$emit('input', newFilter);
      }
```

Same for clearAll I guess